Initial PR for version 1.2

jarmokortetjarvi opened the pull request.

jarmokortetjarvi committed:
Add a placeholder init file

jarmokortetjarvi committed:
Add a gitignore file

jarmokortetjarvi committed:
Add hidden files to the ignore list

jarmokortetjarvi committed:
Add and init file

jarmokortetjarvi committed:
Add account charts version

jarmokortetjarvi committed:
Add an icon

jarmokortetjarvi committed:
Rename the icon

jarmokortetjarvi committed:
Add the account template

jarmokortetjarvi committed:
Add the account types template

jarmokortetjarvi committed:
Add a tax code template

jarmokortetjarvi committed:
Remove an obsolete line

jarmokortetjarvi committed:
Add tax template

jarmokortetjarvi committed:
Add an account chart template

jarmokortetjarvi committed:
Modify the init file commit

jarmokortetjarvi committed:
Fix "TULOS" parent

jarmokortetjarvi committed:
Modify accounts chart accounts and numbering

jarmokortetjarvi committed:
Modify the module name

jarmokortetjarvi committed:
Fix a reference

jarmokortetjarvi committed:
Add a comment

jarmokortetjarvi committed:
Fix the default accounts

jarmokortetjarvi committed:
Add reconciliable to sale profit accounts

jarmokortetjarvi committed:
Fix account type

jarmokortetjarvi committed:
Fix TAVVVV18000A account types

jarmokortetjarvi committed:
Fix TAVVVV18300A account types

jarmokortetjarvi committed:
Fix TAVVVV18400A account types

jarmokortetjarvi committed:
Fix TAVVVV1910 account type ids

jarmokortetjarvi committed:
Fix TAVT account type ids

jarmokortetjarvi committed:
Fix payable types

jarmokortetjarvi committed:
Fix payable reconciliation

jarmokortetjarvi committed:
Fix TULOS account type ids

jarmokortetjarvi committed:
Fix expense type ids

jarmokortetjarvi committed:
Update the version number

jarmokortetjarvi committed:
Fix tax names

jarmokortetjarvi committed:
Fix tax signs

jarmokortetjarvi committed:
Fix Osto ALV 24% tax signs

jarmokortetjarvi committed:
Fix purchase tax signs

jarmokortetjarvi committed:
Fix sales tax signs

jarmokortetjarvi committed:
Fix an account name

jarmokortetjarvi committed:
Fix account names

jarmokortetjarvi committed:
Modify ALV to alv

jarmokortetjarvi committed:
Rollback last commit

jarmokortetjarvi committed:
Fix account names

jarmokortetjarvi committed:
Add tax codes

jarmokortetjarvi committed:
Fix tax codes

jarmokortetjarvi committed:
Fix tax code numbering

jarmokortetjarvi committed:
Fix tax cod numbering

jarmokortetjarvi committed:
Delete obsolete columns

jarmokortetjarvi committed:
Fix the account ids and parents

jarmokortetjarvi committed:
Remove obsolete icons
Add icon to correct path

jarmokortetjarvi committed:
Fix the account references

jarmokortetjarvi committed:
Update the template name and references

jarmokortetjarvi committed:
Update tax codes and ids

jarmokortetjarvi committed:
Update tax template

jarmokortetjarvi committed:
Fix the row order

jarmokortetjarvi committed:
Fix a header

jarmokortetjarvi committed:
Add descriptions

jarmokortetjarvi committed:
Fix a typo

jarmokortetjarvi committed:
Remove erronous description column
Reorder rows

jarmokortetjarvi committed:
Fix the tax codes

jarmokortetjarvi committed:
Remove a helper column

jarmokortetjarvi committed:
Fix the ids

jarmokortetjarvi committed:
Update the references

jarmokortetjarvi committed:
Update the references

jarmokortetjarvi committed:
Modify descriptions

jarmokortetjarvi committed:
Reassign columns

jarmokortetjarvi committed:
Fix the account root

jarmokortetjarvi committed:
Update the version number

jarmokortetjarvi committed:
Fix hierarchy

jarmokortetjarvi committed:
Update "1767 - Maksetut vuokravakuudet" type as regular

jarmokortetjarvi committed:
Add tax sequences for tax statement ordering

jarmokortetjarvi committed:
Add Tawasta to contributors

jarmokortetjarvi committed:
Update description

jarmokortetjarvi committed:
Add readme

jarmokortetjarvi committed:
Improve account names (remove extra semicolons)

jarmokortetjarvi committed:
Merge branch 'master' of https://github.com/Tawasta/l10n-finland

jarmokortetjarvi committed:
Merge branch '8.0' of https://github.com/Tawasta/l10n-finland

Conflicts:
	README.md

jarmokortetjarvi committed:
Improve description

jarmokortetjarvi committed:
Merge pull request #1 from Tawasta/master

Merge branches

jarmokortetjarvi committed:
Add contributors

jarmokortetjarvi committed:
Add a placeholder for maintainer

jarmokortetjarvi committed:
Add a module icon attribution

jarmokortetjarvi committed:
Add missing punctuation

jarmokortetjarvi committed:
Add licence

jarmokortetjarvi committed:
Improve licensing text

jarmokortetjarvi committed:
Add OCA as maintainer

jarmokortetjarvi committed:
Rename README.md to README.rst

jarmokortetjarvi committed:
Add installation header placeholder

jarmokortetjarvi committed:
Add a placeholder for installation instructions

jarmokortetjarvi committed:
Remove redundant licencing text

mlaitinen commented:
Excellent! I'm very happy to see Finnish contributions.

We have been working on similar module https://github.com/avoinsystems/l10n_fi_rapko which is based on [Raportointikoodisto](http://www.raportointikoodisto.fi/). I think we should work together to integrate both of the accounting charts to this repository so that they'd share common data (ie. taxes, fiscal positions).

jarmokortetjarvi commented:
I absolutely agree about working together to build one comprehensive Finnish localization module.

I have examined the repository you linked earlier with some detail, having the integration of both implementations in mind. It pains me to know that the community would already have a very mature l10n_fi, if this version was released in the beginning of the year, when you have worked on your module.

jarmokortetjarvi commented:
The main "problem" with the version in this PR is that it uses the full Liikekirjurit chart of accounts. It could be a good idea to make the main version with less accounts (e.g. http://www.tilit.info/tase_suppea.php and http://www.tilit.info/tuloslaskelma_suppea.php). 

My idea would be to do that by trimming the current chart, preserving all the ids, so it will be possible to upgrade the minimal "default" chart of accounts to the full version later without any conflicts.

dreispt commented:
Thanks for contributing.
The module should be contained in a subdirectory (`l10n_fi_account/` maybe?) and not be at the root of the repository.

jarmokortetjarvi committed:
Move all files to l10n_fi subdirectory

jarmokortetjarvi committed:
Move gitignore to subdir

jarmokortetjarvi committed:
Update readme

jarmokortetjarvi committed:
Add LICENSE text

jarmokortetjarvi committed:
Add OCA dependencies placeholder

mlaitinen commented:
Sorry for bringing this up until now, but I agree with @dreispt. If the repository is called `l10n_fi` then maybe we should put directories like 
- `l10n_fi_account` or `l10n_fi_liikekirjuri` (for @dreispt: liikekirjuri is a common chart of accounts in Finland)
- `l10n_fi_rapko` for Raportointikoodisto (another chart of accounts with its structure designed for official reports)
- `l10n_fi_invoice` for an invoice tempate that fills the Finnish legal requirements for invoices (this is something we're releasing hopefully this week)

IMHO if there's going to be a module called `l10n_fi`, it should contain everything that is essential for every Finnish company using Odoo, eg. address format, payment reference on the invoice, and so on. Or maybe that's something we should contribute directly to Odoo S.A. (?)

jarmokortetjarvi commented:
No problem. This is a great stage to bring out observations and opinions. I actually agree with your suggestion.

I think l10n_fi_account is a better name if we plan to combine the two accounting charts that we currently have. The actual combining, however, may not be possible due different basic structures in the charts.

In other case l10n_fi_liikekirjuri would be a better name.

However, I think we should steer towards one chart of accounts. I grudgingly think it might be better to base it to raportointikoodisto to support it becoming the standard chart of accounts.

jarmokortetjarvi committed:
Rename l10n_fi to l10n_fi_liikekirjuri

dreispt commented on l10n_fi_liikekirjuri/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+    'license': 'AGPL-3',
+    'website': 'http://www.vizucom.com, http://www.tawasta.com',
+    'depends': ['sale_stock', 'report_webkit', 'account'],
+    'description': """
```
You can remove this for the README.rst to be used instead.

dreispt commented on l10n_fi_liikekirjuri/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+The number of accounts is approx. 950
+The number of headers is approx. 300
+""",
+    'depends': ['base_iban', 'base_vat', 'account_chart'],
```
You have two `depends`keys!

dreispt commented on l10n_fi_liikekirjuri/__openerp__.py:
```diff
@@ -0,0 +1,54 @@
+        'data/account.tax.template.csv',
+    ],
+    'installable': 'True',
+    'images': [],
```
deprecated, remove

jarmokortetjarvi committed:
Fix duplicate dependencies

jarmokortetjarvi committed:
Remove redundant description

jarmokortetjarvi committed:
Remove a deprecated value

jarmokortetjarvi committed:
Update the website

dreispt commented:
Please remove pyc files.
You should add repo initialization files, such as `.travis.yml`. 
See https://github.com/OCA/maintainer-quality-tools/tree/master/sample_files

dreispt commented on l10n_fi_liikekirjuri/.gitignore:
```diff
@@ -0,0 +1 @@
+.*
```
Move gitignore to the root.
You may prefer to use the one at https://github.com/OCA/maintainer-quality-tools/tree/master/sample_files

dreispt commented on l10n_fi_liikekirjuri/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+{
+    'name': 'Finland - Accounting',
+    'category': 'Localization/Account Charts',
+    'version': '1.2',
```
8.0.1.2.0
(see https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers)

jarmokortetjarvi committed:
Remove __init__.pyc

dreispt commented:
We need confirmation for the CLA.

jarmokortetjarvi committed:
Move .gitignore to root

jarmokortetjarvi committed:
Improve gitignore with more comprehensive ignore list

Ignore list from
https://github.com/OCA/maintainer-quality-tools/blob/master/sample_files/.gitignore

jarmokortetjarvi commented:
My CLA was confirmed on 26.08.2015

jarmokortetjarvi committed:
Update the version number to match OCA version numbering guidelines

dreispt commented:
Manually verified CLAs are posted on the thread by someone from the board. Can you ask the person that checked it to confirm here?

dreispt commented:
:+1: Looks good! Just need some confirmation of the CLA check (and a second review)

mlaitinen commented:
I have already installed this module in my own environment but unfortunately haven't had the time to review it. I'm trying to get it done this week.

mlaitinen commented:
There are some account names with whitespaces in them, such as "V a s t a t t a v a a" and "O s a k e y h t i ö", which is something you can't find using the Odoo search, unless you know exactly what you are looking for.

The build seems to be failing with the following error:

```
2015-09-01 10:05:03,953 22634 INFO 3114148-1-c3e0b2-all openerp.modules.module: module l10n_fi_liikekirjuri: creating or updating database tables
2015-09-01 10:05:03,953 22634 WARNING 3114148-1-c3e0b2-all openerp.addons.base.module.module: module l10n_fi_liikekirjuri: description is empty !
2015-09-01 10:05:03,953 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.account.type.csv
2015-09-01 10:05:04,080 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.account.template.csv
2015-09-01 10:05:09,601 22634 INFO 3114148-1-c3e0b2-all openerp.modules.loading: module l10n_fi_liikekirjuri: loading data/account.tax.code.template.csv
2015-09-01 10:05:09,603 22634 ERROR 3114148-1-c3e0b2-all openerp: Failed to initialize database `3114148-1-c3e0b2-all`.
Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/cli/server.py", line 97, in preload_registry
    db, registry = openerp.pooler.get_db_and_pool(dbname,update_module=update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/pooler.py", line 33, in get_db_and_pool
    registry = RegistryManager.get(db_name, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/registry.py", line 203, in get
    update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/registry.py", line 233, in new
    openerp.modules.load_modules(registry.db, force_demo, status, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 354, in load_modules
    loaded_modules, update_module)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 256, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 188, in load_module_graph
    load_data(module_name, idref, mode)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 76, in <lambda>
    load_data = lambda *args: _load_data(cr, *args, kind='data')
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/modules/loading.py", line 118, in _load_data
    tools.convert_csv_import(cr, module_name, pathname, fp.read(), idref, mode, noupdate)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3114148-1-c3e0b2/openerp/tools/convert.py", line 935, in convert_csv_import
    raise Exception(_('Module loading %s failed: file %s could not be processed:\n %s') % (module, fname, warning_msg))
Exception: Module loading l10n_fi_liikekirjuri failed: file l10n_fi_liikekirjuri/data/account.tax.code.template.csv could not be processed:
 Line 1 : sequence
```

Getting closer .. :)

dreispt commented:
@jarmokortetjarvi It would help reviews if you added the `.travis.yml` file at the root as I mentioned before.

jarmokortetjarvi committed:
Add .travis.yml for CI

jarmokortetjarvi committed:
Improve obscure account names

jarmokortetjarvi committed:
Fix a too long line

jarmokortetjarvi committed:
Improve author wrapping

jarmokortetjarvi committed:
Update module name

jarmokortetjarvi committed:
Rename the README.rst file to README.md

jarmokortetjarvi committed:
Remove the sequence column

jarmokortetjarvi committed:
Improve the chart of accounts name

jarmokortetjarvi committed:
Fix reference ids

mlaitinen commented:
There's something horribly wrong with the Odoo Runbot. I just rebuilt the latest commit and I got logged in to OpenERP v7 ??

gurneyalex commented:
Hi
I noticed the branches 8.0 and 9.0 were not there in this repository, so I created them. Resubmitting this PR with the correct  target branch should allow runbot to build correctly.

jarmokortetjarvi committed:
Improve tax codes and tax template and fix their references
